spec/unit/pops/issues_spec.rb

```diff
@@ -24,6 +24,25 @@
     expect(x.format()).to eq("Classes, definitions, and nodes may only appear at toplevel or inside other classes")
   end
 
+  it "should format a ILLEGAL_TOP_CONSTRUCT_LOCATION message correctly" do
+    result = <<-RESULT
+Unacceptable location. Unacceptable top level construct(s) at:
+foo
+bar, line: 6
+line: 8
+in file: a/file/path
+RESULT
+    result = result.chomp
+
+    x = Puppet::Pops::Issues::ILLEGAL_TOP_CONSTRUCT_LOCATION
+    expect(x.format(:errors => [
+      ["foo"],
+      ["bar", "line: 6"],
+      ["line: 8"]
+    ],
+    :file => 'a/file/path')).to eq(result)
+  end
```

Comment:
I think the test should be for the resulting actual logged message from an actual found problem instead of this low level test as that reveals if the message contains "in file" more that once etc.